domain/machine/watcher_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package machine_test
+
+import (
+	"context"
+
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/core/watcher/watchertest"
+	"github.com/juju/juju/domain"
+	service "github.com/juju/juju/domain/machine/service"
+	state "github.com/juju/juju/domain/machine/state"
+	changestreamtesting "github.com/juju/juju/internal/changestream/testing"
+	loggertesting "github.com/juju/juju/internal/logger/testing"
+)
+
+type watcherSuite struct {
+	changestreamtesting.ModelSuite
+}
+
+var _ = gc.Suite(&watcherSuite{})
+
+func (s *watcherSuite) TestWatchWithCreate(c *gc.C) {
+	machineService, watcherC := s.machineServiceAndWatcher(c)
+
+	// Create a machine
+	uuid, err := machineService.CreateMachine(context.Background(), "machine-1")
+	c.Assert(err, gc.IsNil)
+	watcherC.AssertChange(uuid)
+}
+
+func (s *watcherSuite) TestWatchWithDelete(c *gc.C) {
+	machineService, watcherC := s.machineServiceAndWatcher(c)
+
+	// Create a machine
+	uuid, err := machineService.CreateMachine(context.Background(), "machine-1")
+	c.Assert(err, gc.IsNil)
+	// Delete the machine
+	err = machineService.DeleteMachine(context.Background(), "machine-1")
+	c.Assert(err, gc.IsNil)
+	watcherC.AssertChange(uuid)
```

Comment:
This isn't quite correct, instead I'd expect one change when we create machine, then one for when we delete the machine.

```suggestion
	// Create a machine
	uuid, err := machineService.CreateMachine(context.Background(), "machine-1")
	c.Assert(err, gc.IsNil)
	
	// Ensure that the changestream is idle.
	s.ModelSuite.AssertChangeStreamIdle(c)
	
	// Assert the first change
	watcherC.AssertChange(uuid)
	
	// Delete the machine
	err = machineService.DeleteMachine(context.Background(), "machine-1")
	c.Assert(err, gc.IsNil)

	// Ensure that the changestream is idle.
	s.ModelSuite.AssertChangeStreamIdle(c)
	
	watcherC.AssertChange(uuid)
```